File: Oauth2Simples/src/main/java/br/com/estudos/oauth2/repository/TypeBalanceRepository.java
package br.com.estudos.oauth2.repository;

import br.com.estudos.oauth2.model.TypeBalance;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import org.springframework.data.jpa.repository.Query;

import java.util.List;

@Repository
public interface TypeBalanceRepository extends CrudRepository<TypeBalance,Long>{
     
     @Query("SELECT tb FROM TypeBalance tb WHERE tb.typeCompany= 'BROKER' AND tb.situation = 'S'")
     List<TypeBalance> findAllByBroker();

}

File: kotlin/kotlinEstudos/src/main/java/br/com/kotlin/estudos/repository/TypeBalanceRepository.kt
package br.com.kotlin.estudos.repository

import br.com.kotlin.estudos.model.repository.financial.Stock
import br.com.kotlin.estudos.model.repository.financial.TypeBalance
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface TypeBalanceRepository: CrudRepository<TypeBalance,Long> {

    fun findByType(type: Int): TypeBalance

}